### PR TITLE
#105: Added noc and l1 alignment variables into flatbuffer

### PR DESCRIPTION
**Pull request by tapspatel**

### CODE REVIEW

**Comment by tapspatel** [comment]
Tested with:

```
cmake --build build -- check-ttmlir

./build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline --ttnn-serialize-to-binary="output=out.ttnn" test/ttmlir/Dialect/TTNN/simple_multiply.mlir
ttrt read out.ttnn &> debug.log

ttrt query --save-system-desc n150.ttsys
ttrt read n150.ttsys
```

File: runtime/lib/ttnn/runtime.cpp [hunk 1]
```diff
@@ -9,6 +9,11 @@
 #include "ttmlir/Target/TTNN/Target.h"
 #include "ttmlir/Version.h"
 
+// references to alignment byte values
+constexpr static const std::uint32_t &REF_NOC_L1_ALIGNMENT = L1_ALIGNMENT;
+constexpr static const std::uint32_t &REF_PCIE_ALIGNMENT = PCIE_ALIGNMENT;
+constexpr static const std::uint32_t &REF_NOC_DRAM_ALIGNMENT = DRAM_ALIGNMENT;
```
Comment (nsmithtt) [inline runtime/lib/ttnn/runtime.cpp:15]: Let's change these to be:
```
constexpr static std::uint32_t kNocL1Alignment = L1_ALIGNMENT;
```

Or just use `L1_ALIGNMENT` and others directly in `CreateChipDesc`.

File: third_party/CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,7 @@ set(TTMETAL_INCLUDE_DIRS
   ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/${ARCH_EXTRA_DIR}
   ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/${ARCH_NAME}
   ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal/tt_eager
+  ${PROJECT_SOURCE_DIR}/third_party/tt-metal/tt_metal/hostdevcommon
```
Comment (nsmithtt) [inline third_party/CMakeLists.txt:27]: Is this needed? I didn't actually see a new include as part of this diff.

**Review by nsmithtt — CHANGES_REQUESTED** [review]

**Review by nsmithtt — APPROVED** [review]
LGTM thank you!

**Pull request merged**